docs/docs/guides/how-to-parameterize-cassandra.md

```diff
@@ -0,0 +1,396 @@
+mkdir kurtosis-cass-cluster && cd kurtosis-cass-cluster
+```
+
+Next, create a [Starlark][starlark] file called `main.star` inside your new directory with the following contents:
+
+```python
+DEFAULT_NUMBER_OF_NODES = 3
+CASSANDRA_NODE_PREFIX= "cassandra-node-"
+CASSANDRA_NODE_IMAGE = "cassandra:4.0"
+
+
+CLUSTER_COMM_PORT_ID = "cluster"
+CLUSTER_COM_PORT_NUMBER =  7000
+CLUSTER_COM_PORT_PROTOCOL = "TCP"
+
+
+CLIENT_COMM_PORT_ID = "client"
+CLIENT_COM_PORT_NUMBER = 9042
+CLIENT_COM_PORT_PROTOCOL = "TCP"
+
+
+FIRST_NODE_INDEX = 0
+  
+def run(plan, args):
+    num_nodes = DEFAULT_NUMBER_OF_NODES
+
+    # Simple check to verify that we have at least 1 node in our cluster
+    if num_nodes == 0:
+       fail("Need at least 1 node to Start Cassandra cluster got 0")
+
+    # Iteratively add each node to the cluster, with the given names and serviceConfig specified below
+    for node in range(0, num_nodes):
+       node_name = get_service_name(node)
+       config = get_service_config(num_nodes)
+       plan.add_service(name = node_name, config = config)
+    
+    node_tool_check = "nodetool status | grep UN | wc -l | tr -d '\n'"
+
+    check_nodes_are_up = ExecRecipe(
+       command = ["/bin/sh", "-c", node_tool_check],
+    )
+
+    # Wait for the nodes to be up and ready to establish connections and receive traffic 
+    plan.wait(
+        service_name = get_first_node_name(),
+        recipe = check_nodes_are_up, 
+        field = "output", 
+        assertion = "==", 
+        target_value = str(num_nodes), 
+        timeout = "8m", 
+        )
+
+    return {"node_names": [get_service_name(x) for x in range(num_nodes)]}
+
+def get_service_name(node_idx):
+    return CASSANDRA_NODE_PREFIX + str(node_idx)
+
+def get_service_config(num_nodes):
+    seeds = ["cassandra-node-"+str(x) for x in range(0, num_nodes)]
+    return ServiceConfig(
+        image = CASSANDRA_NODE_IMAGE,
+        ports = {
+            CLUSTER_COMM_PORT_ID : PortSpec(number = CLUSTER_COM_PORT_NUMBER, transport_protocol = CLUSTER_COM_PORT_PROTOCOL),
+            CLIENT_COMM_PORT_ID : PortSpec(number = CLIENT_COM_PORT_NUMBER, transport_protocol = CLIENT_COM_PORT_PROTOCOL),
+        },
+        env_vars = {
+            "CASSANDRA_SEEDS":",".join(seeds),
+            # without this set Cassandra tries to take 8G and OOMs
+            "MAX_HEAP_SIZE": "512M",
+            "HEAP_NEWSIZE": "1M",
+        }
+    )
+
+def get_first_node_name():
+    return get_service_name(FIRST_NODE_INDEX)
+```
+
+Finally, save your newly created file and, from the working directory we created, run the following command:
+
+```bash
+kurtosis run --enclave cassandra-cluster main.star
+```
+
+:::info
+Kurtosis will run validation checks against our code to ensure that it will work before spinning up the containers for our 3-node Cassandra cluster. We won’t dive into the details of how validation checks are used by Kurtosis in this guide, but you can read more about them [here][multi-phase-runs].
```

Comment:
```suggestion
Kurtosis will run validation checks against your code to ensure that it will work before spinning up the containers for our 3-node Cassandra cluster. We won’t dive into the details of how validation checks are used by Kurtosis in this guide, but you can read more about them [here][multi-phase-runs].
```